src/core/utility/rate.cpp

```diff
@@ -40,7 +40,7 @@ void CRate::Sleep() {
   /* Sleep if necessary */
   if(unMicroSecDiff < m_unNominalPeriod) {
     ::timespec tSleepPeriod;
-    tSleepPeriod.tv_sec = (m_unNominalPeriod - unMicroSecDiff) / 1e6;
+    tSleepPeriod.tv_sec = static_cast<UInt32>((m_unNominalPeriod - unMicroSecDiff) / 1e6);
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Yes--you get a double -> int conversion warning without it